src/components/implementation/nicmgr/dpdk/nicmgr.h

```diff
@@ -38,8 +42,18 @@ struct client_session {
 
 	u32_t ip_addr; 
 	u16_t port;
+	int thd_state;
 
 	struct pkt_ring_buf pkt_ring_buf;
+	struct pkt_ring_buf pkt_tx_ring;
+
+	int tx_init_done;
+	struct sync_sem sem;
+
+	/* number of blocked loops of the tenant, this is counted each time when the thread is blocked */
```

Comment:
....when the thread is blocked when receiving packets (that have not yet arrived).